src/revaultd/mod.rs

```diff
@@ -58,51 +60,56 @@ impl RevaultD {
     fn call<T: Serialize + Debug, U: DeserializeOwned + Debug>(
         &self,
         method: &str,
-        input: T,
-    ) -> Result<U, client::error::Error> {
+        input: Option<T>,
+    ) -> Result<U, RevaultDError> {
         self.client
             .send_request(method, input)
             .and_then(|res| res.into_result())
+            .map_err(|e| match e {
+                client::error::Error::NoErrorOrResult => RevaultDError::NoAnswerError,
+                _ => RevaultDError::RPCError(format!("method {} failed: {}", method, e)),
+            })
     }
 
-    pub fn get_info(&self) -> Result<GetInfoResponse, client::error::Error> {
-        self.call("getinfo", GetInfoRequest {})
+    pub fn get_info(&self) -> Result<GetInfoResponse, RevaultDError> {
+        self.call("getinfo", Option::<Request>::None)
     }
 }
 
-/// getinfo
-
-/// getinfo request
 #[derive(Debug, Clone, Deserialize, Serialize)]
-pub struct GetInfoRequest {}
+struct Request {}
+
+/// getinfo
 
 /// getinfo response
 #[derive(Debug, Clone, Deserialize, Serialize)]
 pub struct GetInfoResponse {
     pub blockheight: u64,
     pub network: String,
-    pub sync: u32,
+    pub sync: f64,
     pub version: String,
 }
 
 // RevaultD can start only if a config path is given.
-pub fn start_daemon(config_path: PathBuf) -> Result<(), RevaultDError> {
+pub async fn start_daemon(config_path: &PathBuf) -> Result<(), RevaultDError> {
```

Comment:
Yep, now that i can test daemon startup i see that it will spawn the daemon thread but will then error with:
> ConnectionRefused

While the daemon is in warmup:
> [2020-12-09][12:31:45][revaultd][INFO] Setting up database

You should probably poll the RPC for 30s (or even more) and eventually fail if it's still not up